docs/docs/tutorials/web_socket_counter.md

```diff
@@ -0,0 +1,410 @@
+import 'package:web_socket_counter/counter/counter.dart';
+
+Handler middleware(Handler handler) => handler.use(counterProvider);
+```
+
+:::info
+For more information, refer to the [middleware documentation](/docs/basics/middleware).
+:::
+
+## Using the Counter
+
+We can access the `CounterCubit` instance from our WebSocket handler via `context.read<CounterCubit>()`.
+
+```dart
+import 'package:dart_frog/dart_frog.dart';
+import 'package:dart_frog_web_socket/dart_frog_web_socket.dart';
+import 'package:web_socket_counter/counter/counter.dart';
+
+Future<Response> onRequest(RequestContext context) async {
+  final handler = webSocketHandler(
+    (channel, protocol) {
+      // A new client has connected to our server.
+      // Subscribe the new client to receive notifications
+      // whenever the cubit state changes.
+      final cubit = context.read<CounterCubit>()..subscribe(channel);
+
+      // Send the current count to the new client.
+      channel.sink.add('${cubit.state}');
+
+      // Listen for messages from the client.
+      channel.stream.listen(
+        (event) {
+          switch (event) {
+            // Handle an increment message.
+            case '__increment__':
+              cubit.increment();
+              break;
+            // Handle a decrement message.
+            case '__decrement__':
+              cubit.decrement();
+              break;
+            // Ignore any other messages.
+            default:
+              break;
+          }
+        },
+        // The client has disconnected.
+        // Unsubscribe the channel.
+        onDone: () => cubit.unsubscribe(channel),
+      );
+    },
+  );
+
+  return handler(context);
+}
+```
+
+First, we subscribe the newly connected client to the `CounterCubit` in order to receive updates whenever the cubit state changes.
+
+Next, we send the current count to the new client via `cubit.state`.
+
+When the client sends a new message, we invoke the `increment`/`decrement` method on the cubit based on the message.
+
+Finally, we unsubscribe the channel when the client disconnects.
+
+:::info
+The `subscribe` and `unsubscribe` APIs are exposed by the `BroadcastCubit` super class from `package:broadcast_bloc`.
+:::
+
+Be sure to save all the changes and hot reload should kick in ⚡️
+
+```
+[hotreload] - Application reloaded.
+```
+
+Now we can update our example script in `example/main.dart`:
+
+```dart
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+void main() async {
+  final channel = WebSocketChannel.connect(Uri.parse('ws://localhost:8080/ws'));
+  channel.stream.listen(print);
+
+  channel.sink.add('__increment__');
+  channel.sink.add('__decrement__');
+
+  channel.sink.close();
+}
+```
+
+Finally, let's run the script:
+
+```
+dart example/main.dart
+```
+
+We should see the following output:
+
+```
+0
+1
+0
+```
+
+:::note
+If you restart the server, the count will always be reset to 0 because it is only maintained in memory.
```

Comment:
```suggestion
If you restart the server, the count will always be reset to 0 because it is only maintained in memory.
```
is this because we're running the example locally?

Reply:
No it's just because the count isn't being persisted anywhere for simplicity.